src/interface_r/R/data_prep.R

```diff
@@ -0,0 +1,102 @@
+#'Prep data for h2o4gpu algorithms
+#'
+#'@param data_table `data.table` object containing data that needs to be preprocessed for h2o4gpu
+#'@param data_path Path to data that needs to be preprocessed for h2o4gpu. Only needed if `data_table` parameter is not provided/available
+#'@param response Response column as a string or index
+#'@param save_as_csv Whether to save processed data as a csv
+#'@param save_csv_path Path to save processed data as a csv
+#'@export
+prep_data <- function(data_table=NULL, data_path=NULL, response=NULL, save_as_csv=FALSE, save_csv_path=NULL){
+  
+  if (save_as_csv && is.null(save_csv_path)) {
+    stop("`save_as_csv` is set to TRUE but `save_csv_path` is NULL. Please specify a path to save processed data")
+  }
+  
+  if (is.null(data_table)) {
+    if (!is.null(data_path)) {
+      print(paste0("Reading in -> ",data_path))
+      DT <- fread(data_path)
+    } else {
+      stop("Parameters `data_table` & `data_path` are NULL. Need to specify at least one.")
+    }
+  } else {
+    DT <- data_table
```

Comment:
Why not making this argument required without default?

Reply:
I will make it required.